module/nvpair/nvpair.c

```diff
@@ -3681,6 +3681,248 @@ nvs_xdr(nvstream_t *nvs, nvlist_t *nvl, char *buf, size_t *buflen)
 	return (err);
 }
 
+#define	NVP(buf, size, len, elem, type, vtype, ptype, format) { \
+	vtype	value; \
+	int rc; \
+\
+	(void) nvpair_value_##type(elem, &value); \
+	rc = snprintf(buf, size, "%*s%s: " format "\n", indent, "", \
+	nvpair_name(elem), (ptype)value); \
+	if (rc < 0) \
+		return (rc); \
+	size = MAX((int)size - rc, 0); \
+	buf = size == 0 ? NULL : buf_end - size; \
```

Comment:
Here and below `buf_end` is not passed as parameter, but seems to be global.

Also at least FreeBSD style recommend `\`'s to be right-justified.